src/lib/utils/fixFingerprintDataTypes.ts

```diff
@@ -0,0 +1,24 @@
+import { IFingerprint } from "../../types";
+import { fingerprintNoQuotes } from "../../types/Fingerprint";
+
+export function fixFingerprintDataTypes(fingerprint: IFingerprint): IFingerprint {
+    const newFp = (fingerprint as unknown) as Record<string, unknown>;
+
+    // set correct data types (int are intigers, bool are booleans)
+    Object.entries(fingerprint).forEach(([key, value]) => {
+        if (value == null || value == "null") {
+            newFp[key] = null;
+        } else if (fingerprintNoQuotes.includes(key) && value.toLowerCase !== undefined) {
+            if (value.toLowerCase() == "false") {
+                newFp[key] = false;
+            } else if (value.toLowerCase() == "true") {
+                newFp[key] = true;
+            } else if (value.includes(".")) {
+                newFp[key] = parseFloat(value);
+            } else {
+                newFp[key] = parseInt(value);
+            }
```

Comment:
We can convert this into a single `if statement`
> Note: `parseFloat` will return an `int` if the value is in fact an integer, for example `"123" -> 123` or `"123.00" -> 123`
```suggestion
            } else if (isNaN(value)) {
                newFp[key] = parseFloat(value);
            }
```